resin/models/environment_variables.py

```diff
@@ -61,57 +54,73 @@ def get_all(self, uuid):
         """
 
         device = self.device.get(uuid)
-        params = {
-            'filter': 'device',
-            'eq': device['id']
-        }
-        return self.base_request.request(
-            'device_environment_variable', 'GET', params=params,
-            endpoint=self.settings.get('pine_endpoint')
-        )['d']
-
-    def create(self, uuid, env_var_name, value):
+        service_installs = self.service_install.get_all_by_device(device['id'])
+
+        env_vars = []
+        for service_install in service_installs:
+            params = {
+                'filter': 'service_install',
+                'eq': service_install['id']
+            }
+
+            env_vars += (self.base_request.request(
+                'device_service_environment_variable', 'GET', params=params,
+                endpoint=self.settings.get('pine_endpoint')
+            )['d'])
+        return env_vars
```

Comment:
This is doing N + 1 requests to get environment variables (where N is the number of services on your device). That's going to be _really_ slow if you have more than a couple of services, and is going to hammer the API. You can do this in one pine query, and we definitely should.

The UI does this one query for example with this: https://github.com/resin-io/resin-ui/blob/250de0c375e28fc2b5c0237ecd092620aefd69a1/src/services/models/environment-model-service.ts#L121-L147